corehq/blobs/tests/test_swiftdb.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import re
+from io import BytesIO
+
+import requests_mock
+from django.test import TestCase
+
+from corehq.blobs.tests.test_fsdb import _BlobDBTests
+from corehq.blobs.tests.util import TemporarySwiftBlobDB
+
+DEFAULT_CONTENT = b'content'
+
+URL = 'http://swift.test/v1/AUTH_mowcdmumusr'
+
+
+def assert_body_equal(test, expected):
+    def _assert(request, context):
+        test.assertEquals(list(request.body), [expected])
+    return _assert
+
+
+def assert_copy_headers(test):
+    def _assert(request, context):
+        test.assertIn('destination', request.headers)
+    return _assert
+
+
+def mock_put(test, mock, content=DEFAULT_CONTENT):
+    mock.put(test._url(match_any=True), text=assert_body_equal(test, content))
+
+
+def mock_get(test, mock, content=DEFAULT_CONTENT):
+    mock.get(test._url(match_any=True), body=BytesIO(DEFAULT_CONTENT), headers={
+        'Content-Length': str(len(content))
+    })
+
+
+def mock_head(test, mock, content=DEFAULT_CONTENT):
+    mock.head(test._url(match_any=True), headers={
+        'Content-Length': str(len(content))
+    })
+
+
+@requests_mock.mock()
+class TestSwiftBlobDB(TestCase, _BlobDBTests):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestSwiftBlobDB, cls).setUpClass()
+        config = {
+            'authurl': 'https://swift.test/auth/v1.0/',
+            'user': 'test',
+            'preauthurl': URL,
+            'preauthtoken': 'token',
+            'container': 'test'
+        }
+        cls.db = TemporarySwiftBlobDB(config)
+
+    def _url(self, object=None, match_any=False):
+        regex = False
+        url = '{}/{}'.format(URL, self.db.container)
+        if match_any:
+            assert not object
+            object = '.*'
+            regex = True
+        if object:
+            url = '{}/{}'.format(url, object)
+        return re.compile(url) if regex else url
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.db.close()
+        super(TestSwiftBlobDB, cls).tearDownClass()
+
+    def test_put_and_get(self, mock):
+        mock_put(self, mock)
+        mock_get(self, mock)
+        super(TestSwiftBlobDB, self).test_put_and_get()
```

Comment:
Do the mocks cause an error to be raised if `connection.put_object(...)` and/or `connection.get_object(...)` are _not_ called by the super method? If not, would it work to simply mock all the operations (get, put, head, delete, etc.) in `self.setUp()`? I think then you would not need to override most of these test methods.